src/pages/studyView/charts/barChart/CustomBinsModal.tsx

```diff
@@ -21,24 +34,42 @@ export default class CustomBinsModal extends React.Component<
 > {
     binSeparator: string = ',';
     @observable private currentBinsValue = '';
+    @observable private currentGenerateBinsConfig: GenerateBinsConfig;
 
     constructor(props: Readonly<ICustomBinsProps>) {
         super(props);
         makeObservable(this);
         if (this.props.currentBins) {
-            this.currentBinsValue = _.sortBy(this.props.currentBins).join(
-                `${this.binSeparator} `
-            );
+            const bins = _.sortBy(this.props.currentBins);
+            this.currentBinsValue = bins.join(`${this.binSeparator} `);
         }
+        this.currentGenerateBinsConfig = this.props.store.generateBinsConfig;
```

Comment:
this doesn't seem quite right. what should be I think is 
(1) you need to initialize currentGenerateBinsConfig as a full object or else @observable won't apply properly, even if you're going to update all its values in the constructor
(2) you need to update its values one by one in the constructor, not update the entire pointer.

Reply:
does that seem right, @alisman ?